core/src/main/java/ai/timefold/solver/core/impl/phase/AbstractPhase.java

```diff
@@ -190,6 +182,7 @@ protected <Score_ extends Score<Score_>> void predictWorkingStepScore(AbstractSt
     @Override
     public void stepEnded(AbstractStepScope<Solution_> stepScope) {
         if (!isNested()) {
+            var solver = stepScope.getPhaseScope().getSolverScope().getSolver();
```

Comment:
To ensure that these solver changes do not cause score corruption, we need to avoid sharing the score director in nested phases, while the solver itself can be shared. Given that the previous statement is correct, the changes will not cause any issues. Does that make sense?